envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ControlPlane.kt

```diff
@@ -280,6 +281,42 @@ class ControlPlane private constructor(
             override fun newThread(r: Runnable) = Thread(r, "$threadNamePrefix-${counter.getAndIncrement()}")
         }
 
+        private fun meterThreadPoolExecutor(
+            corePoolSize: Int,
+            maximumPoolSize: Int,
+            keepAliveTimeMillis: Long,
+            workQueue: BlockingQueue<Runnable>,
+            poolExecutorName: String
+        ): ThreadPoolExecutor {
+            val threadPoolExecutor = ThreadPoolExecutor(
+                corePoolSize,
+                maximumPoolSize,
+                keepAliveTimeMillis,
+                TimeUnit.MILLISECONDS,
+                workQueue,
+                ThreadNamingThreadFactory(poolExecutorName)
+            )
+            meterExecutor(threadPoolExecutor, poolExecutorName)
+            return threadPoolExecutor
+        }
+
+        private fun newMeterFixedThreadPool(executorServiceName: String, poolSize: Int): ExecutorService {
```

Comment:
I think it should be `metered` with `ed`.
With `meter` you add metrics to an existing executor, but you create a new `metered` executor.

Reply:
done